src/lib.rs

```diff
@@ -307,6 +307,16 @@ impl<T: LedgerAccess, const N: usize> Ledger<T, N> {
         &mut self,
         addr: Address<usize, Page>,
         length: Offset<usize, Page>,
+    ) -> Result<(), Error> {
+        self.unmap_with(addr, length, |_| {})
+    }
+
+    /// Delete sub-regions and call a function on each deleted region.
+    pub fn unmap_with(
+        &mut self,
+        addr: Address<usize, Page>,
+        length: Offset<usize, Page>,
+        mut func: impl FnMut(&Record<T>),
```

Comment:
```suggestion
        mut f: impl FnMut(&Record<T>),
```
Just like https://doc.rust-lang.org/nightly/core/option/enum.Option.html#method.map https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.map

Which will also avoid awkward syntax highlighting in GitHub diffs

Does this need to be `FnMut`? Why not a more-generic `FnOnce`?

Reply:
It's not really `map` .. it's `mmap` and this function might mutate state outside.

Reply:
changed the name, though